File: backend/backend/Models/Animal.cs
using backend.Data;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace backend.Models
{
    public class Animal
    {
        [Key]
        public Guid Id { get; set; } = Guid.NewGuid();

        [Required, MaxLength(100)]
        public string Nom { get; set; } = string.Empty;

        [Required, MaxLength(50)]
        public string Espece { get; set; } = string.Empty ;

        [MaxLength(50)]
        public string Race { get; set; } =string.Empty ;

        public int Age { get; set; }

        [Required]
        public string Sexe { get; set; } = string.Empty;

        public string Allergies { get; set; } = string.Empty;
        public string AnttecedentsMedicaux { get; set; } = string.Empty;


        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        public List<Vaccination> Vaccinations { get; set; } = new List<Vaccination>();

        [ForeignKey("AppUser")]
        public Guid OwnerId { get; set; }
        public AppUser Owner { get; set; } = null!;

    }

}

File: frontend/client_app/lib/models/animals_models/animal.dart
class Animal {
  final String id;
  final String name;
  final String espece;
  final String race;
  final int age;
  final String sexe;
  final String allergies;
  final String antecedentsmedicaux;

  Animal({
    required this.id,
    required this.name,
    required this.espece,
    required this.race,
    required this.age,
    required this.sexe,
    required this.allergies,
    required this.antecedentsmedicaux,
  });

  factory Animal.fromJson(Map<String, dynamic> json) {
    return Animal(
      id: json['id'].toString(),
      name: json['name'] ?? '',
      espece: json['espece'] ?? '',
      race: json['race'] ?? '',
      age: json['age'] is int
          ? json['age']
          : int.tryParse(json['age'].toString()) ?? 0,
      sexe: json['sexe'] ?? '',
      allergies: json['allergies'] ?? '',
      antecedentsmedicaux: json['anttecedentsmedicaux'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'espece': espece,
      'race': race,
      'age': age,
      'sexe': sexe,
      'allergies': allergies,
      'anttecedentsmedicaux': antecedentsmedicaux, // lowercase, double 't'
    };
  }
}
